emission/analysis/modelling/similarity/similarity_metric.py

```diff
@@ -27,15 +27,21 @@ def similarity(self, a: List[float], b: List[float]) -> List[float]:
         """
         pass
 
-    def similar(self, a: List[float], b: List[float], thresh: float) -> bool:
+    def similar(self, a: List[float], b: List[float], thresh: float, clusteringWay :str = 'origin-destination') -> bool:
         """compares the features, returning true if they are similar
         within some threshold
 
         :param a: features for a trip
         :param b: features for another trip
         :param thresh: threshold for similarity
+        :clusterinWay: clustering based on origin/destination/origin-destination-pair
         :return: true if the feature similarity is within some threshold
         """
         similarity_values = self.similarity(a, b)
-        is_similar = all(map(lambda sim: sim <= thresh, similarity_values))
+        if clusteringWay == 'origin':
+            is_similar = similarity_values[0] <= thresh
+        elif clusteringWay == 'destination':
+            is_similar = similarity_values[1] <= thresh
+        else:
+            is_similar = all(map(lambda sim: sim <= thresh, similarity_values))
```

Comment:
so the map here is iterating over a two element array?

Reply:
yes. removed it in next commit.